language/vm/src/file_format.rs

```diff
@@ -374,17 +375,36 @@ pub struct FieldDefinition {
     pub signature: TypeSignature,
 }
 
+/// `VisibilityScope` restricts who may call a given function.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum VisibilityScope {
+    /// The function can only be called within the module that defines it.
+    Private,
+    /// The function can be called by any module and script.
+    Public,
+    /// Similar to "private", but in addition, the function can also be called
+    /// by any module specified in the list of module handles.
+    Protected(Vec<ModuleHandleIndex>),
+}
+
+impl Default for VisibilityScope {
+    fn default() -> Self {
+        VisibilityScope::Private
+    }
+}
+
 /// A `FunctionDefinition` is the implementation of a function. It defines
 /// the *prototype* of the function and the function body.
-
 #[derive(Clone, Debug, Default, Eq, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(params = "usize"))]
 pub struct FunctionDefinition {
     /// The prototype of the function (module, name, signature).
     pub function: FunctionHandleIndex,
     /// Flag to indicate if this function is public.
-    pub is_public: bool,
+    pub visibility: VisibilityScope,
```

Comment:
Another clarification is that: the description of this PR shows how the bytecode layout looks like. The actual field in the `FunctionDefinition` Rust struct is here.